src/main/java/core/basesyntax/impl/CustomStorage.java

```
@@ -0,0 +1,89 @@
+package core.basesyntax.impl;
+
+@SuppressWarnings("unchecked")
+public class CustomStorage<K, V> {
+    private static final int INIT_CAPACITY = 10;
+    private int size = 0;
+    private final Entry<K, V>[] table;
+
+    static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        public K getKey() {
+            return key;
+        }
+
+        public void setKey(K key) {
+            this.key = key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+    }
+
+    public CustomStorage() {
+        table = (Entry<K, V>[]) new Entry[INIT_CAPACITY];
+    }
+
+    public V get(K key) {
+        for (Entry<K, V> entry : table) {
+            K entryKey = entry != null ? entry.getKey() : null;
+            if (entry != null
+                    && (entryKey == key || entryKey != null
+                    && entryKey.equals(key))) {
+                return entry.getValue();
+            }
+        }
+        return null;
+    }
+
+    public void put(K key, V value) {
+        int keyIndex = keyIndex(key);
+
+        if (this.size >= INIT_CAPACITY && keyIndex != -1) {
```

This condition for checking if the storage is full is incorrect. It should throw an exception only when you try to add a **new** element to a full storage. The current logic prevents **updating** an existing element when the storage is full, which should be a valid operation.